src/hypothesis/strategies.py

```diff
@@ -627,19 +627,30 @@ def streaming(elements):
 @defines_strategy
 def characters(whitelist_categories=None, blacklist_categories=None,
                blacklist_characters=None, min_codepoint=None,
-               max_codepoint=None):
+               max_codepoint=None, whitelist_characters=None):
     """Generates unicode text type (unicode on python 2, str on python 3)
     characters following specified filtering rules.
 
-    This strategy accepts lists of Unicode categories, characters of which
-    should (`whitelist_categories`) or should not (`blacklist_categories`)
-    be produced.
+    When no filtering rules are specifed, any character can be produced.
 
-    Also there could be applied limitation by minimal and maximal produced
-    code point of the characters.
+    If ``min_codepoint`` or ``max_codepoint`` is specifed, then only
+    characters having a codepoint in that range will be produced.
 
-    If you know what exactly characters you don't want to be produced,
-    pass them with `blacklist_characters` argument.
+    If ``whitelist_categories`` is specified, then only characters from those
+    Unicode categories will be produced. This is a further restriction,
+    characters must also satisfy `min_codepoint` and `max_codepoint`.
+
+    If ``blacklist_categories`` is specified, then any character from those
+    categories will not be produced. This is a further restriction,
+    characters that match both ``whitelist_categories`` and
+    ``blacklist_categories`` will not be produced.
+
+    If ``whitelist_characters`` is specified, then any additional characters
+    in that list will also be produced.
+
+    If ``blacklist_characters`` is specified, then any characters in that list
+    will be not be produced. Characters in both ``whitelist_characters`` and
+    ``blacklist_characters`` will not be produced.
```

Comment:
This would be inconsistent with how `whitelist_categories` and `blacklist_categories` interact, e.g. `characters(whitelist_categories=('Lu', 'Nd'), blacklist_categories('Nd'))` is *not* an error. Are you okay with that?